sdk-extensions/incubator/src/main/java/io/opentelemetry/sdk/extension/incubator/fileconfig/ResourceDetectorFactory.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.sdk.extension.incubator.fileconfig;
+
+import static java.util.stream.Collectors.joining;
+
+import io.opentelemetry.api.incubator.config.DeclarativeConfigException;
+import io.opentelemetry.sdk.autoconfigure.internal.SpiHelper;
+import io.opentelemetry.sdk.extension.incubator.fileconfig.internal.model.ExperimentalResourceDetectorModel;
+import io.opentelemetry.sdk.resources.Resource;
+import java.io.Closeable;
+import java.util.List;
+import java.util.Map;
+
+final class ResourceDetectorFactory
+    implements Factory<ExperimentalResourceDetectorModel, Resource> {
+
+  private static final ResourceDetectorFactory INSTANCE = new ResourceDetectorFactory();
+
+  private ResourceDetectorFactory() {}
+
+  static ResourceDetectorFactory getInstance() {
+    return INSTANCE;
+  }
+
+  @Override
+  public Resource create(
+      ExperimentalResourceDetectorModel model, SpiHelper spiHelper, List<Closeable> closeables) {
+    Map<String, Object> additionalProperties = model.getAdditionalProperties();
+    if (!additionalProperties.isEmpty()) {
```

Comment:
nit: invert the `if` and throw early, the rest of this then gets unintended.

Reply:
Much better